lib/Controller/PollController.php

```diff
@@ -91,33 +98,66 @@ public function createPoll(string $question, array $options, int $resultMode, in
 				$question,
 				$options,
 				$resultMode,
-				$maxVotes
+				$maxVotes,
+				$draft,
 			);
 		} catch (\Exception $e) {
 			$this->logger->error('Error creating poll', ['exception' => $e]);
 			return new DataResponse([], Http::STATUS_BAD_REQUEST);
 		}
 
-		$message = json_encode([
-			'message' => 'object_shared',
-			'parameters' => [
-				'objectType' => 'talk-poll',
-				'objectId' => $poll->getId(),
-				'metaData' => [
-					'type' => 'talk-poll',
-					'id' => $poll->getId(),
-					'name' => $question,
-				]
-			],
-		], JSON_THROW_ON_ERROR);
+		if (!$draft) {
+			$message = json_encode([
+				'message' => 'object_shared',
+				'parameters' => [
+					'objectType' => 'talk-poll',
+					'objectId' => $poll->getId(),
+					'metaData' => [
+						'type' => 'talk-poll',
+						'id' => $poll->getId(),
+						'name' => $question,
+					]
+				],
+			], JSON_THROW_ON_ERROR);
 
-		try {
-			$this->chatManager->addSystemMessage($this->room, $attendee->getActorType(), $attendee->getActorId(), $message, $this->timeFactory->getDateTime(), true);
-		} catch (\Exception $e) {
-			$this->logger->error($e->getMessage(), ['exception' => $e]);
+			try {
+				$this->chatManager->addSystemMessage($this->room, $attendee->getActorType(), $attendee->getActorId(), $message, $this->timeFactory->getDateTime(), true);
+			} catch (\Exception $e) {
+				$this->logger->error($e->getMessage(), ['exception' => $e]);
+			}
 		}
 
-		return new DataResponse($this->renderPoll($poll, []), Http::STATUS_CREATED);
+		return new DataResponse($this->renderPoll($poll), Http::STATUS_CREATED);
```

Comment:
This returns `Poll` in both cases, not the `PollDraft`

Reply:
I'll fix that